pgml-dashboard/notebooks/static/notebooks/js/notebook.js

```diff
@@ -227,7 +230,6 @@ export default class extends Controller {
     const entries = []
 
     for (let entry of form.entries()) {
-      if (entry[0] === 'csrfmiddlewaretoken')
```

Comment:
Make sure not to include the textarea input by accident, it's empty, the contents come from CodeMirror (see code below)...

Reply:
If you want to use the form as-is, issue `this.newCellCodeMirror.save()` before collecting data from the form.

Reply:
Is that why you're not using a normal hotwire form submit/render cycle?

Reply:
That and also hotwire re-renders the whole page which resets the scrolling position making it hard to work with the notebook with multiple cells.

Reply:
I think that can be fixed with individual controllers per cell rather than one controller for the full notebook